src/memory_manager.rs

```diff
@@ -419,6 +561,43 @@ impl<M: Memory> MemoryManagerInner<M> {
         // Ceiling division.
         (num_pages + self.bucket_size_in_pages as u64 - 1) / self.bucket_size_in_pages as u64
     }
+
+    fn free(&mut self, id: MemoryId) {
+        self.memory_sizes_in_pages[id.0 as usize] = 0;
+        let buckets = self.memory_buckets.remove(&id);
+        if let Some(vec_buckets) = buckets {
+            for bucket in vec_buckets {
+                self.unallocated_buckets.push_front(bucket);
+                self.allocated_buckets -= 1;
+            }
```

Comment:
Let's say we have two memories:

```
Memory 1: Buckets[1, 2, 3],
Memory 2: Buckets[4, 5, 6],
```

Then I do the following:

```
free(Memory 1)
free(Memory 2)

get(Memory 3)
grow(Memory 3, 3)
```

IIUC the code above will first use buckets 4, 5, 6 before using 1, 2, 3. This doesn't matter too much right now, but it might matter if in the future we add defragmentation. It's more desirable to use the buckets in order (i.e. 1, 2, 3, then 4, 5, 6. Does that make sense?

Reply:
Just because something is mentioned in a research paper doesn't mean we should do it :) Using smaller bucket IDs for us allows to minimize our memory footprint compared to using larger bucket IDs.